src/engine/BMDie.php

```diff
@@ -138,7 +138,7 @@ public function init($sides, array $skills = NULL) {
             $this->max = 0;
         } else {
             $this->min = 1;
-            $this->max = $sides;
+            $this->max = (int)$sides;
```

Comment:
shadowshade: why was this change necessary?  I don't like it because we agreed awhile ago that the correct approach is to convert numbers from strings to ints at the outside limits of the site --- when things are loaded from the DB, and when things arrive at the API layer from the network.  (There's an issue about this somewhere; i can look it up if need be.)  So why are we casting this sides value to an int deep in the guts of BMDie initialization, rather than making sure it's an int whenever it's defined, and not getting as far as BMDie initialization if it isn't?

Reply:
I believe you're talking about #712. I don't remember why I found it necessary to do the cast, but I'll take a look and see if I can shift that back up to BMGame. It won't be able to go any further than that because the frontend is sending the an arbitrary recipe string, which we need to parse with the backend.